src/patch.rs

```diff
@@ -290,14 +290,13 @@ fn add_patches_for_packages(
             .entry(&p.name)
             .or_insert(Item::Value(Value::InlineTable(Default::default())))
             .as_inline_table_mut()
-            .ok_or_else(|| format!("Patch entry for `{}` isn't an inline table!", p.name))?;
+            .ok_or(format!("Patch entry for `{}` isn't an inline table!", p.name))?;
 
-        let path = if p.manifest_path.ends_with("Cargo.toml") {
+        if p.manifest_path.ends_with("Cargo.toml") {
             p.manifest_path.pop();
-            p.manifest_path
-        } else {
-            p.manifest_path
-        };
+        }
+
+        let path: PathBuf = p.manifest_path.into();
```

Comment:
```suggestion
        let path = p.manifest_path;
```

Reply:
The compiler disagrees with your suggestion. We need the type to be known for the call before L304 to be OK with calling `.display(...)`.